test cases/frameworks/10 gtk-doc/meson.build

```diff
@@ -31,9 +31,15 @@ libfoo = shared_library('foo', 'foo.c',
   dependencies: gobject,
 )
 
+deps = []
+if host_machine.system() == 'darwin'
+  deps += dependency('appleframeworks', modules : ['Foundation', 'CoreFoundation'])
```

Comment:
This does not seem to reproduce the issue on CI. And since I do not have access to a Darwin machine, I am not sure how to make fail.

Reply:
If the test is being skipped, this is probably due to the macOS CI not having gtkdoc installed?

The truth is, we install surprisingly little in `.github/workflows/macos.yml`

Reply:
Yeah, installing gtkdoc allows me to reproduce it on CI: https://github.com/mesonbuild/meson/runs/5709205241?check_suite_focus=true#step:5:1331